File: app/src/main/java/practicaltask/riseapps/com/practicaltask/ui/countryList/CountryListView.java
package practicaltask.riseapps.com.practicaltask.ui.countryList;

import java.util.List;

import practicaltask.riseapps.com.practicaltask.data.model.CountryDTO;
import practicaltask.riseapps.com.practicaltask.ui.base.BaseView;

interface CountryListView extends BaseView {

    void showProgress();

    void hideProgress();

    void checkInternetConnection();

    void showNoInternetConnectionError();

    void showCountryList(List<CountryDTO> countryDTOS);

    void showDataExecutionError();
}

File: app/src/main/java/practicaltask/riseapps/com/practicaltask/ui/countryList/CountryListView.kt
package practicaltask.riseapps.com.practicaltask.ui.countryList

import practicaltask.riseapps.com.practicaltask.data.model.CountryDTO
import practicaltask.riseapps.com.practicaltask.ui.base.BaseView

interface CountryListView : BaseView {

    fun showProgress()

    fun hideProgress()

    fun checkInternetConnection()

    fun showNoInternetConnectionError()

    fun showCountryList(countryDTOS: List<CountryDTO>)

    fun showDataExecutionError()

    fun showCountryDetail(countryDTO: CountryDTO)
}
